instrumentation/grpc/src/main/java/brave/grpc/TracingClientInterceptor.java

```diff
@@ -93,10 +111,62 @@
     }
   }
 
+  static final class TraceContextCallListener<RespT>
+    extends SimpleForwardingClientCallListener<RespT> {
+    final CurrentTraceContext currentTraceContext;
+    @Nullable final TraceContext invocationContext;
+
+    TraceContextCallListener(
+      Listener<RespT> delegate,
+      CurrentTraceContext currentTraceContext,
+      @Nullable TraceContext invocationContext
+    ) {
+      super(delegate);
+      this.currentTraceContext = currentTraceContext;
+      this.invocationContext = invocationContext;
+    }
+
+    @Override public void onReady() {
+      Scope scope = currentTraceContext.maybeScope(invocationContext);
+      try {
```

Comment:
Not sure if gRPC being Java 8 means we would prefer try/resources here but just in case

Reply:
I didn't move over the retrolambda comment. we still support gRPC 1.2 :P

Reply:
I will make a comment in the pom.xml in next change which is that when we bring the floor version up to v1.16.0, we can bump to Java 1.7 and use try/resources naturally (unless retrolambda still complains)

Reply:
wait that's already there.. I'll just add the comment :)